app/models/tag.rb

```diff
@@ -0,0 +1,7 @@
+class Tag < ActiveRecord::Base
+  validates :tagname, presence: true
+  validates_uniqueness_of :tagname
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.